build_daemon/lib/data/server_log.dart

```diff
@@ -2,16 +2,94 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'package:built_collection/built_collection.dart';
 import 'package:built_value/built_value.dart';
 import 'package:built_value/serializer.dart';
+import 'package:logging/logging.dart' as logging;
 
 part 'server_log.g.dart';
 
+/// Logging levels, these have a 1:1 mapping with the levels from
+/// `package:logging`.
+class Level extends EnumClass with Comparable<Level> {
```

Comment:
This is better IMO but man... that's a lot of boilerplate :(

Reply:
Ya, it does have some advantages, like allowing more robust comparison and simplifying the translations between the types since it has an actual name field. Also the name won't ever get mangled or anything. :man_shrugging: